docs/deploy-rbd.md

```diff
@@ -415,6 +415,33 @@ the AWS KMS is expected to contain:
 This Secret is expected to be created by the tenant/user in each namespace where
 Ceph-CSI is used to create encrypted rbd volumes.
 
+#### Configuring KMIP KMS
+
+The Key Management Interoperability Protocol (KMIP) is an extensible communication protocol that defines message formats for the manipulation of cryptographic keys on a key management server.
+Ceph-CSI can be configured to connect to various KMS servers using 
+[KMIP](https://en.wikipedia.org/wiki/Key_Management_Interoperability_Protocol) for encrypting RBD volumes.
+
+There are a few settings that need to be included in the [KMS configuration
+file](../examples/kms/vault/kms-config.yaml):
+
+1. `KMS_PROVIDER`: should be set to `kmip`.
+2. `KMIP_ENDPOINT` KMIP endpoint address 
+3. `KMIP_SECRET_NAME`(optional): name of the Kubernetes Secret which contains the
+   credentials for communicating with KMIP server, defaults to
+   `ceph-csi-kmip-credentials`.
+4. `TLS_SERVER_NAME`(optional): The endpoint server name. Useful when the KMIP endpoint does not have a DNS entry.
+5. `READ_TIMEOUT`(optional): Network read timeout, in seconds. The default value is 10.
+6. `WRITE_TIMEOUT`(optional): Network write timeout, in seconds. The default value is 10.
+
+
+The [Secret with credentials](../examples/kms/vault/kmip-credentials.yaml) for
+the KMIP KMS is expected to contain:
+
+1. `CA_CERT`: CA certificate that will be used to connect to KMIP server.
+2. `CLIENT_CERT`: Client certificate that will be used to connect to KMIP server.
+3. `CLIENT_KEY`: Client key that will be used to connect to KMIP server
+4. `UNIQUE_IDENTIFIER`: Unique ID of the key to use for encrypting/decrypting.
```

Comment:
I am assuming `UNIQUE_IDENTIFIER` is for encrypting/decrypting DEK.
But, do we want this identifier as an input from the UI ? Can't we do the same as we are doing for HPCS ?

Reply:
> IIUC `UNIQUE_IDENTIFIER` is for encrypting/decrypting DEK.
> But, do we want this identifier as an input from the UI ? Can't we do the same as we are doing for HPCS ?

@SanjalKatiyar 
Yes this needs to be an created in the secret, 
We are doing it exactly as in hpcs.
This field is similar to 
https://github.com/ceph/ceph-csi/blob/2fc10ded656c4f587b6a9763f4af3b5147b28b01/examples/kms/vault/kp-credentials.yaml#L11

Reply:
If I am correct, customer_root_key was something that was supported/required by HPCS KMS instance itself...
but in our case we will not be passing `UNIQUE_IDENTIFIER` to the KMS instance, so we are introducing new parameter which customers do not know about (I mean not supported by Cipher Trust Manager itself)...
Is it okay to do so (we can discuss this in tomorrow meeting once as well !!) ?

Reply:
marking it resolved based on discussion